bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/logic/BiweeklyPresentableCalendar.java

```diff
@@ -439,8 +431,47 @@ public Event toEvent() {
             final Summary eventSummary = vEvent.getSummary();
             final String title = eventSummary != null ? eventSummary.getValue() : "-";
             final Description eventDescription = vEvent.getDescription();
-            final String description = eventDescription != null ? eventDescription.getValue() : "";
-            return new Event(title, start, end, description);
+            final String description = eventDescription != null ? eventDescription.getValue() : null;
+            final Location eventLocation = vEvent.getLocation();
+            final String location = eventLocation != null ? eventLocation.getValue() : null;
+            final List<Contact> eventContacts = vEvent.getContacts();
+            final List<String> contactStrings = new ArrayList<>(eventContacts.size());
+            eventContacts.forEach((c) -> contactStrings.add(c.getValue()));
+            final String contact = !contactStrings.isEmpty() ? String.join("; ", contactStrings) : null;
+            final List<Comment> eventComments = vEvent.getComments();
+            final List<String> commentStrings = new ArrayList<>(eventComments.size());
+            eventComments.forEach((c) -> commentStrings.add(c.getValue()));
```

Comment:
How about just `final List<String> commentStrings = vEvent.getComments().stream().map(c -> c.getValue()).toList();`? This way, the local variable `eventComments` wouldn't even be necessary.

This could be taken even further by joining with the subsequent string concatenation, but I'm not sure if this extension is more readable due to the null handling.
```
final String comment = vEvent.getComments().stream().map(c -> c.getValue())
    .collect(Collectors.collectingAndThen(Collectors.joining("\n"), s -> s.isEmpty() ? null : s));
```

Reply:
OK.

Reply:
Readability is one issue with that shortening. I tried to implement your suggestion, it does not compile because of impossible NonNull-Handling (i would have to set a `@Nullable` inside the lambda. I've now split into two statements setting `null` if the string is empty (which would happen as there would be nothing to join). I think thats a good mixture and is better to read also. Please resolve if you think the same way.